azurelinuxagent/common/protocol/hostplugin.py

```diff
@@ -96,26 +104,28 @@ def put_vm_status(self, status_blob, sas_url):
         if status_blob is None or status_blob.vm_status is None:
             logger.error("no status data was provided")
             return
-        url = URI_FORMAT_PUT_VM_STATUS.format(self.endpoint, HOST_PLUGIN_PORT)
-        logger.verbose("Posting VM status to host channel [{0}]".format(url))
-        status = textutil.b64encode(status_blob.data)
-        headers = {HEADER_VERSION: API_VERSION,
-                   "Content-type": "application/json",
-                   HEADER_CONTAINER_ID: self.container_id,
-                   HEADER_HOST_CONFIG_NAME: self.role_config_name}
-        blob_headers = [{'headerName': 'x-ms-version',
-                         'headerValue': status_blob.__storage_version__},
-                        {'headerName': 'x-ms-blob-type',
-                         'headerValue': status_blob.type}]
-        data = json.dumps({'requestUri': sas_url, 'headers': blob_headers,
-                           'content': status}, sort_keys=True)
         try:
-            response = restutil.http_put(url, data, headers)
+            url = URI_FORMAT_PUT_VM_STATUS.format(self.endpoint, HOST_PLUGIN_PORT)
+            logger.verbose("Posting VM status to host plugin")
+            status = textutil.b64encode(status_blob.data)
+            blob_type = status_blob.type if status_blob.type else config_blob_type
```

Comment:
I agree this could be improved, but I think the correct change is to remove `get_blob_type` since the `HEAD` call is not actually necessary anymore. For this PR I did not want to make more changes than necessary, but if you feel we should include that let me know.